### PR TITLE
Cody: Add E2E test setup based on Playwright

### PR DESCRIPTION
**Pull request by philipp-spiess**
This PR adds a Playwright test setup to Cody.

**How is this different from the existing integration test?**

Our existing integration tests already download and run VS Code and allow us to create assertions inside the extension host runtime. However, these tests are written on a lower level without easy ways to interact with the UI.

Playwright is a very popular tool with exactly that in mind: interact with the app as if you were a user. This way we can write very high-level tests like this:

```ts
test('requires a valid auth token and allows logouts', async ({ page, sidebar }) => {
    await sidebar.getByRole('textbox', { name: 'Access Token (docs)' }).fill(VALID_TOKEN)
    await sidebar.getByRole('button', { name: 'Sign In' }).click()

    await expect(sidebar.getByText("Hello! I'm Cody.")).toBeVisible()
})
```

**Why Playwright?**

Playwright is the spiritual successor to Puppeteer and adds a lot of nice APIs. It's also used to test various extensions from Microsoft and VS Code itself. 

There's also a lot of insane tooling that comes with using the Playwright test runner that also works when it's attached to an Electron app. Check out this example where I use the `--debug` mode to debug selectors and even record new one based on interaction with the UI (there are some not-so-smooth edges right now, e.g. like it opening an empty Chromium tab in the example below 😅).

https://github.com/sourcegraph/sourcegraph/assets/458591/2e765ef0-e1a1-4e4c-94c0-2f269c42d52a

## Test plan

- This adds a new test which passes

### CODE REVIEW

**Comment by sourcegraph-buildkite** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (+0.00 kb) | 0.00% (+0.52 kb) | 0.00% (+0.52 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/cody-e2e/compare-report.html;tab=live_object) for a full comparison between the commits f28612a and 6d71359 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by eseliger — APPROVED** [review]

File: client/cody/test/e2e/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    src = "tsconfig.json",
+    deps = [
+        "//client/cody:tsconfig",
+    ],
+)
+
+ts_project(
+    name = "e2e_tests",
+    testonly = True,
+    srcs = [
```
Comment (valerybugakov) [inline client/cody/test/e2e/BUILD.bazel:23]:
```suggestion
    testonly = True,
    tsconfig = ":tsconfig",
    srcs = [
```

**Review by valerybugakov — APPROVED** [review]
`bazel tests` fail because Bazel cannot find the input file we specified in the `BUILD.bazel` file:

```
(21:05:09) ERROR: /buildkite/builds/buildkite-agent-bazel-69478b967b-7xqxc-1/sourcegraph/sourcegraph/client/cody/test/e2e/BUILD.bazel:12:10: Copying file client/cody/test/e2e/tsconfig.json failed: missing input file '//client/cody/test/e2e:tsconfig.json'
````

This expectation is defined in `client/cody/test/e2e/BUILD.bazel` by the `tsconfig ` target. We either want to use the `tsconfig.json` from the package level — `client/cody/tsconfig.json` or need to create a `tsconfig` in `client/cody/test/e2e`.

**Review by abeatrix — APPROVED** [review]
You're a hero Philipp 🦸‍♂️

**Review by mrnugget — APPROVED** [review]
Very cool!

File: client/cody/test/e2e/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+# npm_link_all_packages(name = "node_modules")
+
+# gazelle:js_resolve vscode //:node_modules/@vscode
+# gazelle:js_files **/*.{ts,tsx}
+# gazelle:js_test_files **/*.{ts,tsx}
+
+ts_config(
+    name = "tsconfig",
+    src = "../../tsconfig.json",
```
Comment (valerybugakov) [inline client/cody/test/e2e/BUILD.bazel:14]: We have to use the "absolute path" here. For that, we need to have a BUILD.bazel file in the root of the test directory.

We can keep one tsconfig.json file for the `client/cody/test` subtree:
```
client/cody/test
  tsconfig.json
  BUILD.bazel
```

File: client/cody/test/integration/.eslintrc.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
 // @ts-check
 
-const baseConfig = require('../.eslintrc')
+const baseConfig = require('../../.eslintrc')
 module.exports = {
-  extends: '../.eslintrc.js',
+  extends: '../../.eslintrc.js',
```
Comment (valerybugakov) [inline client/cody/test/integration/.eslintrc.js:5]: Can we drop this `.eslintrc.js`?

Comment (philipp-spiess) [reply client/cody/test/integration/.eslintrc.js:5]: Can we just have both subdirectories being linted via the `client/cody` config?

Comment (valerybugakov) [reply client/cody/test/integration/.eslintrc.js:5]: Sure, that's why I do in my PR.

**Comment by philipp-spiess** [comment]
Screenshot 2023-05-19 at 17 55 06

🥳

**Review by abeatrix — APPROVED** [review]

**Pull request merged**